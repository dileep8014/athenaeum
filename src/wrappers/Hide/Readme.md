### Hide Example:

```jsx
    <div>
      <Text>hello, </Text>
      <Hide hideOn='small medium'>
        <Spacer small />
      </Hide>
      <Text>world</Text>
    </div>
```
