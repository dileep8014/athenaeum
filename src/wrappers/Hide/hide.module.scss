$screens: (
  $small-only: small,
  $medium-only: medium,
  $large-only: large,
  $xlarge-only: xLarge,
  $xxlarge-only: xxLarge
);

@each $screenSize, $className in $screens {
  @media #{$screenSize} {
    .#{$className} {
      display: none;
    }
  }
}

