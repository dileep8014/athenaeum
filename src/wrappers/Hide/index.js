import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './hide.module.scss';

function getHideList(hideOn) {
  return hideOn.split(' ').map(hide => styles[hide]);
}

function Hide(props) {
  const {
    children,
    className,
    hideOn,
  } = props;

  const hideList = getHideList(hideOn);

  const classes = [
    styles['hide'],
    className,
    ...hideList,
  ];

  return (
    <div className={classnames(...classes)}>
      { children }
    </div>
  );
}


Hide.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
  /**
   * The screen sizes on which to hide the children -- use screen sizes from `Layout` component
   */
  hideOn: PropTypes.string.isRequired,
};

export default Hide;
