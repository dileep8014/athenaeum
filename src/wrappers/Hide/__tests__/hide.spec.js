import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Hide from 'wrappers/Hide';

describe('<Hide />', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Hide hideOn='small'>hello</Hide>);

    expect(wrapper.type()).toEqual(Hide);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<Hide hideOn='small'>hello</Hide>).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
