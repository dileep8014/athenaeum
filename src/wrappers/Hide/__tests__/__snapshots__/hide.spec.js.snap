// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hide /> renders correctly 1`] = `
<div
  className="hide mobile"
>
  hello
</div>
`;
